### ISSUE TITLE
Error code LNK2019

### ISSUE THREAD

**Issue by rubygobay** [eclipse/paho.mqtt.cpp#121]
![capture](https://user-images.githubusercontent.com/30035408/28560680-e2096762-7145-11e7-961f-0955130408ae.PNG)
see that

**Comment by fpagliughi** [comment]
Those are calls in the Paho C Synchronous Library. The C++ wrapper uses the Async version of the C library, so I'm not sure how those calls are being pulled in. Can you most some more information?

**Comment by rubygobay** [comment]
`#include "stdafx.h"
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "MQTTClient.h"
#define TIMEOUT     10000L
int main(int argc, char* argv[])
{
	MQTTClient client;
	MQTTClient_connectOptions conn_opts = MQTTClient_connectOptions_initializer;
	MQTTClient_message pubmsg = MQTTClient_message_initializer;
	MQTTClient_deliveryToken token;
	int rc;
	MQTTClient_create(&client, "tcp://192.168.56.101:1883", "ExampleClientPub",
		MQTTCLIENT_PERSISTENCE_NONE, NULL);
	conn_opts.keepAliveInterval = 20;
	conn_opts.cleansession = 1;
	if ((rc = MQTTClient_connect(client, &conn_opts)) != MQTTCLIENT_SUCCESS)
	{
		printf("Failed to connect, return code %d\n", rc);
		exit(EXIT_FAILURE);
	}
	pubmsg.payload = "Hello World!";
	pubmsg.payloadlen = strlen("Hello World!");
	pubmsg.qos = 1;
	pubmsg.retained = 0;
	MQTTClient_publishMessage(client, "topic", &pubmsg, &token);
	printf("Waiting for up to %d seconds for publication of %s\n"
		"on topic %s for client with ClientID: %s\n",
		(int)(TIMEOUT / 1000), "Hello World!", "topic", "ExampleClientPub");
	rc = MQTTClient_waitForCompletion(client, token, TIMEOUT);
	printf("Message with delivery token %d delivered\n", token);
	MQTTClient_disconnect(client, 10000);
	MQTTClient_destroy(&client);
	return rc;
}`

here my code, I'm using MQTTClient.h library

**Comment by rubygobay** [comment]
```
> #include "stdafx.h"
> #include <stdio.h>
> #include <stdlib.h>
> #include <string.h>
> #include "MQTTClient.h"
> #define TIMEOUT     10000L
> int main(int argc, char* argv[])
> {
> 	MQTTClient client;
> 	MQTTClient_connectOptions conn_opts = MQTTClient_connectOptions_initializer;
> 	MQTTClient_message pubmsg = MQTTClient_message_initializer;
> 	MQTTClient_deliveryToken token;
> 	int rc;
> 	MQTTClient_create(&client, "tcp://192.168.56.101:1883", "ExampleClientPub",
> 		MQTTCLIENT_PERSISTENCE_NONE, NULL);
> 	conn_opts.keepAliveInterval = 20;
> 	conn_opts.cleansession = 1;
> 	if ((rc = MQTTClient_connect(client, &conn_opts)) != MQTTCLIENT_SUCCESS)
> 	{
> 		printf("Failed to connect, return code %d\n", rc);
> 		exit(EXIT_FAILURE);
> 	}
> 	pubmsg.payload = "Hello World!";
> 	pubmsg.payloadlen = strlen("Hello World!");
> 	pubmsg.qos = 1;
> 	pubmsg.retained = 0;
> 	MQTTClient_publishMessage(client, "topic", &pubmsg, &token);
> 	printf("Waiting for up to %d seconds for publication of %s\n"
> 		"on topic %s for client with ClientID: %s\n",
> 		(int)(TIMEOUT / 1000), "Hello World!", "topic", "ExampleClientPub");
> 	rc = MQTTClient_waitForCompletion(client, token, TIMEOUT);
> 	printf("Message with delivery token %d delivered\n", token);
> 	MQTTClient_disconnect(client, 10000);
> 	MQTTClient_destroy(&client);
> 	return rc;
> }


``` 

here my code , I'm using MQTTClient.h library

**Comment by rubygobay** [comment]
I'm trying mqtt communication using c ++

**Comment by rubygobay** [comment]
when i'm using this script then there was an error and this is the notification of error meren ok
> // async_consume.cpp
> //
> // This is a Paho MQTT C++ client, sample application.
> //
> // This application is an MQTT consumer/subscriber using the C++
> // asynchronous client interface, employing the  to receive messages
> // and status updates.
> //
> // The sample demonstrates:
> //  - Connecting to an MQTT server/broker.
> //  - Subscribing to a topic
> //  - Receiving messages through the synchronous queuing API
> //
> 
> /*******************************************************************************
>  * Copyright (c) 2013-2017 Frank Pagliughi <fpagliughi@mindspring.com>
>  *
>  * All rights reserved. This program and the accompanying materials
>  * are made available under the terms of the Eclipse Public License v1.0
>  * and Eclipse Distribution License v1.0 which accompany this distribution.
>  *
>  * The Eclipse Public License is available at
>  *    http://www.eclipse.org/legal/epl-v10.html
>  * and the Eclipse Distribution License is available at
>  *   http://www.eclipse.org/org/documents/edl-v10.php.
>  *
>  * Contributors:
>  *    Frank Pagliughi - initial implementation and documentation
>  *******************************************************************************/
> 
> #include <iostream>
> #include <cstdlib>
> #include <string>
> #include <cstring>
> #include <cctype>
> #include <thread>
> #include <chrono>
> #include "mqtt/async_client.h"
> 
> using namespace std;
> 
> const string SERVER_ADDRESS	{ "tcp://192.168.56.101:1883" };
> const string CLIENT_ID		{ "async_consume" };
> const string TOPIC 			{ "topic" };
> 
> const int  QOS = 1;
> 
> /////////////////////////////////////////////////////////////////////////////
> 
> int main(int argc, char* argv[])
> {
> 	mqtt::connect_options connOpts;
> 	connOpts.set_keep_alive_interval(20);
> 	connOpts.set_clean_session(true);
> 
> 	mqtt::async_client cli(SERVER_ADDRESS, CLIENT_ID);
> 
> 	try {
> 		cout << "Connecting to the MQTT server..." << flush;
> 		cli.connect(connOpts)->wait();
> 		cli.start_consuming();
> 		cli.subscribe(TOPIC, QOS)->wait();
> 		cout << "OK" << endl;
> 
> 		// Consume messages
> 
> 		while (true) {
> 			auto msg = cli.consume_message();
> 			if (!msg) break;
> 			cout << msg->get_topic() << ": " << msg->to_string() << endl;
> 		}
> 
> 		// Disconnect
> 
> 		cout << "\nShutting down and disconnecting from the MQTT server..." << flush;
> 		cli.unsubscribe(TOPIC)->wait();
> 		cli.stop_consuming();
> 		cli.disconnect()->wait();
> 		cout << "OK" << endl;
> 	}
> 	catch (const mqtt::exception& exc) {
> 		cerr << exc.what() << endl;
> 		return 1;
> 	}
> 
>  	return 0;
> }
> 
>

**Comment by BeyondAE** [comment]
And i'm in same issue..

![_2017_09_07_15_56_24_890](https://user-images.githubusercontent.com/27330229/30150161-4e176ae6-93e5-11e7-9bb0-7e25d5abf0a6.png)

**Comment by MattiasEppler** [comment]
https://github.com/eclipse/paho.mqtt.c/issues/356

**Comment by MattiasEppler** [comment]
I changed the plattform from x86 to x64 in my testapplication.

But then I get the linker error:
fatal error LNK1112: module machine type 'x86' conflicts with target machine type x64'
paho-mqttpp3.lib(message.obj)

Problem: The cmake file generates visual studio soultion with win32 plattform set.
So I added x64 manualy and tried to compile paho-mqttpp3.lib but then I get same linker error in there,


fatal error LNK1112: module machine type 'x64' conflicts with target machine type x86'
async_client.obj

Any idea how to solve?

**Comment by fpagliughi** [comment]
You need to compile the Paho C library and the C++ library for the same platform. I just tried it with the CMake command line.

For the default build, you can do:
```
mkdir build
cd build
cmake ..
cmake --build .
```
For the x86_64 build, use -G to specify the full compiler with Win64 specified:
```
cmake -G "Visual Studio 14 Win64" ..
cmake --build .
```
As long as you use the same build settings for both projects, it seems to work.

**Comment by fpagliughi** [comment]
As far as I can tell, this is now fixed with the v1.0.1 release.
